### DISCUSSION TITLE
No response received from deepgram streaming websockets

### DISCUSSION THREAD

**Discussion by Mrashidpoor** [deepgram/community#502] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

Hi, I am using the deepgram streaming api and want to transcribe phone calls live while the agents are speaking on the phone. I successfully create the client and connect to the websocket endpoint and send the raw audio data on the call, but I get no responses back when listening for the Transcript event. Below is my part of code handling the event and below is one of the many request ids I have that had no responses.

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

f717ed11-ce57-487b-9819-dbb1408617dc

### If possible, please attach your code or paste it into the text box.

const deepgram = createClient(process.env.DEEPGRAM_API_KEY);
        const live = deepgram.listen.live({
            model: "nova",
            punctuate: true,
            smart_format: true,
            filler_words: true,
            numerals: true,
            keywords: keywords,
            interim_results: true,
            // endpointing: endpointingDelay,
            // utterance_end_ms: utteranceDelay,
            encoding: 'mulaw',
            sample_rate: 8000,
        });

live.on(LiveTranscriptionEvents.Open, () => {
            sttNotifer.emit('connection-opened', live)
            console.log('Stt Connection opened')
            let isStartEventEmitted = false
            let partialTranscript = "";

            live.on(LiveTranscriptionEvents.Transcript, (data) => {
                console.log(data)
                if (!isStartEventEmitted) {
                    sttNotifer.emit("caller-speaking");
                    console.log('caller started speaking')
                    isStartEventEmitted = true
                }
                partialTranscript = data.channel.alternatives[0].transcript ?? ''
                console.log(partialTranscript)

                const speechFinal = data.speech_final
                const type = data.type

                if (speechFinal || type === 'UtteranceEnd') {
                    const finalTranscript = data.channel.alternatives[0].transcript
                    sttNotifer.emit("caller-stopped", finalTranscript);
                    console.log(`Caller: ${finalTranscript}`);
                    isStartEventEmitted = false;
                }
            });
            live.on(LiveTranscriptionEvents.Close, () => {
                console.log('STT connection closed...');
            });
            live.on(LiveTranscriptionEvents.Error, (error) => {
                console.error(error.message);
            });
            live.on(LiveTranscriptionEvents.Warning, (warning) => {
                console.warn(warning);
            });
        });

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment]
Can you also share the part of your code where you sent the data to Deepgram?

I have one guess right now - you mention you are transcribing phone calls, and 8000 Hz mulaw is typical for telephony, but providers like Twilio (for example) don't stream raw audio, their audio is base64 encoded and packed into objects, so it needs some manipulation before sending to Deepgram.

**Reply by Mrashidpoor** [reply]
Thanks Nicola for getting back to me. I was using google stt before this and the wer was very high so I had to switch. Below you can find  my code that sends the raw audio data to deepgram using the sdk.
````
if (message.event === "media" && message.media.track === "inbound") {
          // recognizeStream.write(message.media.payload);
          if (live && live.getReadyState() === 1) {
            live.send(message.media.payload)

          }
        }
````
As you can see in my code, I used to write the raw audio data to google recognition stream. And btw, yes I am using twilio as my telephony provider.

**Reply by Mrashidpoor** [reply]
Hi again Nicola, 
I solved the problem by decoding the audio with the line below:

let decodedAudio = Buffer.from(message.media.payload, 'base64');

You might want to include this in the docs though to avoid confusion for others.

**Reply by nikolawhallon** [reply, marked as answer]
Thanks for the code - this makes sense! That is streaming the base64-encoded mulaw, not the raw audio. Google happens to handle base64-encoded mulaw specifically, probably for exactly these use-cases, but it's not a raw audio format that a media library would recognize. Luckily it should be a one-line fix, just decode the payload. For an example, see this blog post (it's Python though): https://deepgram.com/learn/deepgram-twilio-streaming

The relevant lines are:
```
     if data['event'] == 'media':
      media = data['media']
      chunk = base64.b64decode(media['payload'])
```

Ah, we do have a node example, if that's helpful: https://github.com/deepgram/public-code-examples/tree/master/twilio/node

The relevant lines are:
```
      if (twilioMessage["event"] === "media") {
        const media = twilioMessage["media"];
        const audio = Buffer.from(media["payload"], "base64");
        connection.send(audio);
      }
```

Hopefully this helps!
